### ISSUE TITLE
[CI] MinimumMasterNodesIT tests failed periodically

### ISSUE THREAD

**Issue by dnhatn** [elastic/elasticsearch#27762]
Some tests in `MinimumMasterNodesIT` started failing recently.

1. `MinimumMasterNodesIT#testCanNotPublishWithoutMinMastNodes`: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+periodic/1038/console and https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/993/console

2. `MinimumMasterNodesIT#testMultipleNodesShutdownNonMasterNodes` : https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+oracle-periodic/1642/console and https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+java9-periodic/1645/console

3. `MinimumMasterNodesIT#testCanNotBringClusterDown` : https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+aggressive-opts/482/console

4. `MinimumMasterNodesIT#testDynamicUpdateMinimumMasterNodes`: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+multijob-unix-compatibility/os=ubuntu/421/console and https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/280/console

**Comment by dnhatn** [comment]
Another instance https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+periodic/1037/console

**Comment by ywelsch** [comment]
I've looked at one of the test failures: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+periodic/1038/console

The reason the test fails is that the target ports used for pinging do not match the ports of the nodes that are bound.

The logs show that node_t0, node_t1 and node_t2 bind to ports 9586, 9587 and 9588, respectively:

```
  1> [2017-12-12T07:19:02,123][DEBUG][o.e.t.MockTcpTransport   ] [node_t0] binding server bootstrap to: [::1, 127.0.0.1]
  1> [2017-12-12T07:19:02,143][DEBUG][o.e.t.MockTcpTransport   ] [node_t0] Bound profile [default] to address {[::1]:9586}
  1> [2017-12-12T07:19:02,150][DEBUG][o.e.t.MockTcpTransport   ] [node_t0] Bound profile [default] to address {127.0.0.1:9586}
  1> [2017-12-12T07:19:02,151][INFO ][o.e.t.t.MockTransportService] [node_t0] publish_address {127.0.0.1:9586}, bound_addresses {[::1]:9586}, {127.0.0.1:9586}

  1> [2017-12-12T07:19:02,135][DEBUG][o.e.t.MockTcpTransport   ] [node_t1] binding server bootstrap to: [::1, 127.0.0.1]
  1> [2017-12-12T07:19:02,150][DEBUG][o.e.t.MockTcpTransport   ] [node_t1] Bound profile [default] to address {[::1]:9587}
  1> [2017-12-12T07:19:02,176][DEBUG][o.e.t.MockTcpTransport   ] [node_t1] Bound profile [default] to address {127.0.0.1:9587}
  1> [2017-12-12T07:19:02,176][INFO ][o.e.t.t.MockTransportService] [node_t1] publish_address {127.0.0.1:9587}, bound_addresses {[::1]:9587}, {127.0.0.1:9587}

  1> [2017-12-12T07:19:02,143][DEBUG][o.e.t.MockTcpTransport   ] [node_t2] binding server bootstrap to: [::1, 127.0.0.1]
  1> [2017-12-12T07:19:02,175][DEBUG][o.e.t.MockTcpTransport   ] [node_t2] Bound profile [default] to address {[::1]:9588}
  1> [2017-12-12T07:19:02,179][DEBUG][o.e.t.MockTcpTransport   ] [node_t2] Bound profile [default] to address {127.0.0.1:9588}
  1> [2017-12-12T07:19:02,179][INFO ][o.e.t.t.MockTransportService] [node_t2] publish_address {127.0.0.1:9588}, bound_addresses {[::1]:9588}, {127.0.0.1:9588}
```

For pinging, they use the ports 9580 - 9584, however:

```
  1> [2017-12-12T07:19:02,227][TRACE][o.e.d.z.UnicastZenPing   ] [node_t0] resolved host [127.0.0.1] to [127.0.0.1:9580, 127.0.0.1:9581, 127.0.0.1:9582, 127.0.0.1:9583, 127.0.0.1:9584]
  1> [2017-12-12T07:19:02,227][TRACE][o.e.d.z.UnicastZenPing   ] [node_t0] resolved host [[::1]] to [[::1]:9580, [::1]:9581, [::1]:9582, [::1]:9583, [::1]:9584]
  
  1> [2017-12-12T07:19:02,215][TRACE][o.e.d.z.UnicastZenPing   ] [node_t1] resolved host [127.0.0.1] to [127.0.0.1:9580, 127.0.0.1:9581, 127.0.0.1:9582, 127.0.0.1:9583, 127.0.0.1:9584]
  1> [2017-12-12T07:19:02,215][TRACE][o.e.d.z.UnicastZenPing   ] [node_t1] resolved host [[::1]] to [[::1]:9580, [::1]:9581, [::1]:9582, [::1]:9583, [::1]:9584]
  
  1> [2017-12-12T07:19:02,239][TRACE][o.e.d.z.UnicastZenPing   ] [node_t2] resolved host [127.0.0.1] to [127.0.0.1:9580, 127.0.0.1:9581, 127.0.0.1:9582, 127.0.0.1:9583, 127.0.0.1:9584]
  1> [2017-12-12T07:19:02,239][TRACE][o.e.d.z.UnicastZenPing   ] [node_t2] resolved host [[::1]] to [[::1]:9580, [::1]:9581, [::1]:9582, [::1]:9583, [::1]:9584]
```

What happens is that `InternalTestCluster` configures the three nodes to bind in the range `9580-9600`. I'm not sure why the nodes cannot bind to the sub-range `9580-9585`, something else seems to occupy those. For pinging, we then use the range `9580-9584` (by default the first 5 ports from beginning of the range on).

I don't see which recent PR could be causing this issue. I wonder if it could be caused by just having more integration tests run in a single JVM? The sixth integration test that runs within the same JVM reuses the same port range from the first integration test (i.e. every test increments the port range by 20, modulo 100). It's possibly that one of the earlier integration tests did not properly clean up / free the ports?

Note that we have been running quite a few integration tests in this same JVM:

```
2> NOTE: All tests run in this JVM: [IngestProcessorNotInstalledOnAllNodesIT, IndicesStatsBlocksIT, UpdateIT, ConcurrentDocumentOperationIT, SimpleIndexStateIT, MasterDisruptionIT, RecoveryFromGatewayIT, ShardInfoIT, SnapshotBlocksIT, MatchedQueriesIT, TypesExistsIT, DateHistogramOffsetIT, SearchWhileCreatingIndexIT, SimilarityIT, CustomQueryParserIT, HDRPercentilesIT, LongTermsIT, IndexTemplateBlocksIT, FiltersAggsRewriteIT, IndexActionIT, ParentFieldLoadingIT, BulkProcessorIT, IndexSortIT, GeoDistanceSortBuilderIT, ClusterHealthIT, PreBuiltAnalyzerIntegrationIT, RangeIT, MinBucketIT, SearchSliceIT, DestructiveOperationsIT, DedicatedMasterGetFieldMappingIT, ClusterSearchShardsIT, QueryProfilerIT, IndexAliasesIT, AdjacencyMatrixIT, SearchWithRandomIOExceptionsIT, HistogramIT, ClusterServiceIT, ShardStateIT, GetSettingsBlocksIT, SimpleRoutingIT, SearchScrollWithFailingNodesIT, RandomExceptionCircuitBreakerIT, NodeDisconnectIT, StringTermsIT, GeoBoundsIT, SimpleDataNodesIT, SourceFetchingIT, GeoShapeIntegrationIT, TemplateUpgradeServiceIT, ClusterRerouteIT, DeleteIndexBlocksIT, ValueCountIT, FunctionScoreFieldValueIT, ContextCompletionSuggestSearchIT, IndexShardIT, AckClusterUpdateSettingsIT, RecoveryWithUnsupportedIndicesIT, ForceMergeBlocksIT, ZenDiscoveryIT, ClearIndicesCacheBlocksIT, ScriptedMetricIT, InternalEngineMergeIT, DerivativeIT, SearchFieldsIT, NaNSortingIT, ClusterAllocationExplainIT, MinimumMasterNodesIT]
```
